app/controllers/matches_controller.rb

```diff
@@ -31,25 +30,22 @@ def delete
 
   private
 
-  def adoption_params
-    params.permit(:pet_id, :adopter_account_id, :match_id)
+  def match_params
+    params.require(:match).permit(:pet_id, :adopter_account_id)
   end
 
-  # set status on all applications for a pet
-  def set_statuses_to_adoption_made
-    @applications = Pet.find(params[:pet_id]).adopter_applications
-    @applications.each do |app|
-      unless app.status == "withdrawn"
-        app.status = "adoption_made"
-        app.save
-      end
-    end
+  def set_pet
+    @pet = Pet.find(match_params[:pet_id])
+  end
+
+  def set_match
+    @match = Match.find(params[:id])
   end
 
   def get_pet_id
     return params[:pet_id] if params[:pet_id]
 
-    Match.find(params[:match_id]).pet_id
+    Match.find(params[:id]).pet_id
```

Comment:
Where is this id coming from again?

Reply:
The id is the id passed to the `MatchesController` action. Like for example, if you wrote: 
```ruby
<%= button_to "Delete Match", match_path(@match), method: :delete %>
```
The `id` of `@match` would be given to params. It actually would not exist for actions like `create` where there is no existing `@match`, but in those cases it should be getting the `pet_id` from the early return statement. 

I removed this method in the authz branch actually. It's only used in `same_organization?` in this controller, and that logic has been moved to the policy in that branch.